packages/@aws-cdk/aws-ivs-alpha/lib/recording-configuration.ts

```diff
@@ -0,0 +1,366 @@
+import { CfnRecordingConfiguration } from 'aws-cdk-lib/aws-ivs';
+import { IBucket } from 'aws-cdk-lib/aws-s3';
+import { Duration, Fn, IResource, Resource, Stack, Token } from 'aws-cdk-lib/core';
+import { Construct } from 'constructs';
+
+/**
+ * Properties of the IVS Recording configuration
+ */
+export interface RecordingConfigurationProps {
+  /**
+   * S3 bucket where recorded videos will be stored.
+   */
+  readonly bucket: IBucket;
+
+  /**
+   * The name of the Recording configuration.
+   * The value does not need to be unique.
+   *
+   * @default - auto generate
+   */
+  readonly recordingConfigurationName?: string;
+
+  /**
+   * If a broadcast disconnects and then reconnects within the specified interval,
+   * the multiple streams will be considered a single broadcast and merged together.
+   *
+   * `recordingReconnectWindow` must be between 0 and 300 seconds
+   *
+   * @default - 0 seconds (means disabled)
+   */
+  readonly recordingReconnectWindow?: Duration;
+
+  /**
+   * The set of renditions are recorded for a stream.
+   * For BASIC channels, the CUSTOM value has no effect.
+   * Custom renditions do not apply to BASIC channels.
+   *
+   * @default RenditionSelection.ALL
+   */
+  readonly renditionSelection?: RenditionSelection;
+
+  /**
+   * A list of which renditions are recorded for a stream.
+   * This property must be set when `renditionSelection` is `RenditionSelection.CUSTOM`.
+
+   * @default - no resolution selected
+   */
+  readonly renditions?: Resolution[];
```

Comment:
I would prefer to avoid having this as a separate property that is dependent on the value of `renditionSelection` and instead prefer to use an enum-like class for the type of `renditionSelection` where if  we set it to `RenditionSelection.Custom()` then it would take in a `Resolution[]` as a required parameter` and if we set `RenditionSelection.All()` then it would not include a parameter for `renditions` making it impossible to set the invalid pairing of properties.

Reply:
I've created `RenditionConfiguration` class.